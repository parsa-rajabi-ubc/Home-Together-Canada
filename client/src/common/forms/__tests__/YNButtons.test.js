/**
 * @Author:     Jeff Hatton
 * @Created:    2020.11.21
 *
 * @Description: Yes/No radio button input Component Snapshot test
 *
 */

import React from 'react';
import renderer from  'react-test-renderer'
import YNButton from "../YNButtons";

jest.mock("react-tooltip/node_modules/uuid", () => ({
            v4: () => "00000000-0000-0000-0000-000000000000"}
    )
);

describe('YNButtons', () => {
    describe('Snapshot test', () => {
        it("should render correctly regardless of properties", () => {
            // Given
            const label = 'testLabelString1';
            const name = 'testName'
            // when
            const component = renderer.create(<YNButton label={label} name={name}/>).toJSON();

            // then
            expect(component).toMatchSnapshot();
        });
    })
})