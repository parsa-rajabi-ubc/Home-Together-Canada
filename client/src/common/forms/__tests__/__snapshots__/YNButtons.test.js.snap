// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YNButtons Snapshot test should render correctly regardless of properties 1`] = `
<div>
  <div>
    <label
      className="label text-base"
    >
      testLabelString1
    </label>
    
  </div>
  <label
    className="text-base text-gray-700 pr-8"
  >
    <input
      className="form-radio mt-2 ml-2 mr-1 mb-4"
      name="testName"
      type="radio"
      value="no"
    />
    No
  </label>
  <label
    className="text-base text-gray-700 pr-8"
  >
    <input
      className="form-radio mt-2 ml-2 mr-1 mb-4"
      name="testName"
      type="radio"
      value="yes"
    />
    Yes
  </label>
</div>
`;
